api/client/swarm/init.go

```diff
@@ -36,7 +37,8 @@ func newInitCommand(dockerCli *client.DockerCli) *cobra.Command {
 	}
 
 	flags := cmd.Flags()
-	flags.Var(&opts.listenAddr, "listen-addr", "Listen address")
+	flags.Var(&opts.listenAddr, flagListenAddr, "Listen address")
+	flags.StringVar(&opts.listenInterface, flagListenInterface, "", "Listen on an interface's address")
```

Comment:
Let's be consistent with `--cluster-advertise`.

`--listen-addr=192.168.1.1:4567`
`--listen-addr=eth0:4567`

This happens not to need any engine-api changes.